core/src/main/java/hudson/tasks/ArtifactArchiver.java

```diff
@@ -89,6 +89,9 @@
      * Archive only if build is successful, skip archiving on failed builds.
      */
     private boolean onlyIfSuccessful;
```

Comment:
Should probably be `transient Boolean` so it doesn't get persisted again, and there should be `readResolve` that transforms the loaded value into corresponding `onSuccessful`/`onFailed`/`onAborted` values. This apparently has no support for loading existing configurations (and behaving correctly) at all.